### PR TITLE
[RA-1 Ch03] Added content for Section 3.3.2. Tenant Separation.

### LINKED ISSUE #634: [RA-1 Ch03] 3.3.2. Tenant Separation | Need Content
[RA-1 Ch03] 3.3.2. Tenant Separation | Need Content

### PR DESCRIPTION
**Pull request by msidana**
Fixes #634

### CODE REVIEW

**Comment by msidana** [comment]
> Very good. A question I have is how do we want to recommend separation or do we take the view that it is up to the operator?

The separation at some level is offered by OpenStack (for example different vlans in ovs). For separation like AggregateMultiTenancyIsolation, it is upto operator. At what level the operator wants it - use various schedule filters, create customised(proprietary) scheduler filters etc.
So my view is to keep it to operator on what extent of separation is required.

**Comment by msidana** [comment]
@iangardner22 @pgoyal01 Pls review.

**Comment by msidana** [comment]
@rabiabdel Pls review.

**Comment by rabi-abdel** [comment]
Suggestion to move discussion started on issueL: https://github.com/cntt-n/CNTT/issues/666 into here.

File: doc/ref_arch/openstack/chapters/chapter03.md [hunk 1]
```diff
@@ -185,7 +185,8 @@ This section describes the core set of services and service components needed to
 
 <a name="3.3.2"></a>
 ### 3.3.2. Tenant Isolation
-Content to be developed
+In OpenStack user interfaces and documentation, a group of users is referred to as a project or tenant. With Keystone v1 and v2(both deprecated), the term "tenant" was used. With Keystone v3, the term "project" got adopted and both the terms became interchangeable. However, as CNTT RA uses Keystone v3 in [this](https://cntt-n.github.io/CNTT/doc/ref_arch/openstack/chapters/chapter05.html#5.2) section, so it is recommended to use the term "project" when referring to OpenStack and use [tenant](https://github.com/cntt-n/CNTT/blob/master/doc/ref_model/chapters/chapter03.md#321-tenant) when referring to multi-tenancy . According to [OpenStack glossary](https://docs.openstack.org/doc-contrib-guide/common/glossary.html), Projects represent the base unit of "ownership" in OpenStack, in that all resources in OpenStack should be owned by a specific project.
```
Comment (rabi-abdel) [inline doc/ref_arch/openstack/chapters/chapter03.md]: @msidana please don't use absolute paths, use relative ones. i.e. ../../etc

Comment (msidana) [reply doc/ref_arch/openstack/chapters/chapter03.md]: Fixed.

**Comment by msidana** [comment]
> Does this PR also fixes #666 ? or shall there be a different PR for that one?

@rabi-abdel  Refer my comment https://github.com/cntt-n/CNTT/issues/666#issuecomment-558627532

File: doc/ref_arch/openstack/chapters/chapter03.md [hunk 2]
```diff
@@ -185,7 +185,8 @@ This section describes the core set of services and service components needed to
 
 <a name="3.3.2"></a>
 ### 3.3.2. Tenant Isolation
-Content to be developed
+In OpenStack user interfaces and documentation, a group of users is referred to as a project or tenant. With Keystone v1 and v2(both deprecated), the term "tenant" was used. With Keystone v3, the term "project" got adopted and both the terms became interchangeable. However, as CNTT RA uses Keystone v3 in [this](chapter05.md#5.2) section, so it is recommended to use the term "project" when referring to OpenStack and use [tenant](../../ref_model/chapters/chapter03.md#321-tenant) when referring to multi-tenancy . According to [OpenStack glossary](https://docs.openstack.org/doc-contrib-guide/common/glossary.html), Projects represent the base unit of "ownership" in OpenStack, in that all resources in OpenStack should be owned by a specific project.
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter03.md]: OpenStack Glossary
user: In OpenStack Identity, entities represent individual API consumers and are owned by a specific domain. In OpenStack Compute, a user can be **associated** with roles, projects, or both.

 "In OpenStack user interfaces and documentation, a group of users is referred to as a project or tenant." -- can we delete the sentence as it causes unnecessary confusion as a project is defined later in the paragraph.

File: doc/ref_arch/openstack/chapters/chapter03.md [hunk 3]
```diff
@@ -185,7 +185,8 @@ This section describes the core set of services and service components needed to
...
+OpenStack offers multi-tenancy by means of resource(compute, network and storage)separation via projects. OpenStack offers ways to share virtual resources between tenants while maintaining logical separation. As an example, traffic separation is provided by creating different VLAN ids for different tenant neutron networks. As another example, if host separation is needed, nova scheduler offers AggregateMultiTenancyIsolation scheduler filter to separate tenants in host aggregates. Thus, if a host in an aggregate is configured for a particular tenant, only the instances from that tenant are placed on the host. Overall, tenant isolation ensures that the resources of a project are not affected by resources of another project.
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter03.md]: Can the term "project" be used consistently -- replace "tenant" with "project" except in phrases such as "tenant isolation."

File: doc/ref_arch/openstack/chapters/chapter03.md (hunk 2 above)
Comment (iangardner22) [reply doc/ref_arch/openstack/chapters/chapter03.md]: agreed

Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter03.md]: "... Projects represent the base unit of "ownership" in OpenStack ..." --> "... Projects represent the base unit of **resource (compute, network and storage)** "ownership" in OpenStack ..."

"...  in that **all resources** in OpenStack should be owned by **a specific project**" --> "...  in that all assigned **resources** in OpenStack are owned by **projects**"

Comment (msidana) [reply doc/ref_arch/openstack/chapters/chapter03.md]: Fixed

File: doc/ref_arch/openstack/chapters/chapter03.md (hunk 3 above)
Comment (msidana) [reply doc/ref_arch/openstack/chapters/chapter03.md]: Fixed, thanks !

File: doc/ref_arch/openstack/chapters/chapter03.md (hunk 2 above)
Comment (msidana) [reply doc/ref_arch/openstack/chapters/chapter03.md]: Ok

**Comment by rabi-abdel** [comment]
@pgoyal01 once you approve it as WSL, I will approve and merge.

**Comment by pgoyal01** [comment]
@msidana Can you please make some format changes: a space between word and parentheses {"... word(....)" or "... (...)word" } except where you have a URL. Thanks.

**Comment by msidana** [comment]
> @msidana Can you please make some format changes: a space between word and parentheses {"... word(....)" or "... (...)word" } except where you have a URL. Thanks.

Fixed.

**Comment by pgoyal01** [comment]
@msidana I think this also fixes Issue #666 -- if you agree please close Issue #666 with "Fixed in PR #662." Thanks

**Pull request merged**